include/unifex/sync_wait.hpp

```diff
@@ -96,12 +96,14 @@ struct _receiver {
 
     friend auto
     tag_invoke(tag_t<get_scheduler>, const type& r) noexcept {
-      return r.ctx_.get_scheduler();
+      return r.ctx_->get_scheduler();
```

Comment:
`assert(r.ctx_ != nullptr);` ?

Or do we need to allow this to be called but have it fail if anything is scheduled on the returned scheduler?

Reply:
That is interesting... given that ctx is runtime information, presumably we have to have a runtime failure case here.

Do we have any other operations that can fail to provide useful information via tag_invoke?

Reply:
I've changed it to assert and it does pass the current tests. Something that is guaranteed to complete inline presumably is not going to get_scheduler.

That doesn't feel right, though. It could easily be brittle if there is any chance that scheduler_provider can fail to be correct at runtime.

Returning an Inline scheduler here makes no sense either, allowing delegation to an inline scheduler would be a bad contract.

Maybe scheduler_provider should have get_scheduler return an optional scheduler?
Or we pair it with a has_scheduler operation?

Reply:
Discussed a scheduler wrapper approach that would send an error on submit if the real one was unavailable.